tests/extension/oneapi_device_global/device_global_functional_variables_with_same_name.cpp

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides functional test for device_global
+//
+//  Tests that device_global instance, created in unnamed namespace can be
+//  shadowed by a variable with the same name.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include "../../common/type_coverage.h"
+#include "device_global_common.h"
+#include "type_pack.h"
+
+#define TEST_NAME device_global_functional_variables_with_same_name
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+using namespace device_global_common_functions;
+
+#if defined(SYCL_EXT_ONEAPI_PROPERTY_LIST) && \
+    defined(SYCL_EXT_ONEAPI_DEVICE_GLOBAL)
+namespace oneapi = sycl::ext::oneapi;
+
+namespace variables_with_same_name {
+
+template <typename T>
+struct kernel;
+
+template <typename T>
+oneapi::device_global<T> dev_global;
+
+namespace {
+template <typename T>
+oneapi::device_global<T> dev_global;
+
+/**
+ * @brief Get the device_global instance from unnamed namespace
+ */
+template <typename T>
+oneapi::device_global<T>& get_dg_from_namespace() {
+  return dev_global<T>;
+}
+}  // namespace
+
+template <typename T>
+void run_test(util::logger& log, const std::string& type_name) {
+  bool is_changed_correctly{false};
+  T def_val{};
+  T new_val{};
+  value_helper<T>::change_val(new_val);
+  auto queue = util::get_cts_object::queue();
+  {
+    sycl::buffer<bool, 1> is_changed_corr_buf(&is_changed_correctly,
+                                              sycl::range<1>(1));
+    queue.submit([&](sycl::handler& cgh) {
+      auto is_changed_corr_acc =
+          is_changed_corr_buf.template get_access<sycl::access_mode::write>(
+              cgh);
```

Comment:
At some point it is possible to use SYCL 2020 syntax
```suggestion
     sycl::accessor is_changed_corr_acc { is_changed_corr_buf, cgh };
```

Reply:
Which access mode accessor will have in this case? `sycl::access_mode::read_write` ?

Reply:
Yes. This might be too much. Otherwise add the `write_only` tag.